rust/osm2lanes/src/tags.rs

```diff
@@ -156,10 +165,23 @@ impl TagsRead for Tags {
 pub trait TagsWrite {
     /// Returns the old value of this key, if it was already present.
     fn insert<K: Into<TagKey>, V: Into<String>>(&mut self, k: K, v: V) -> Option<String>;
+    fn checked_insert<K: Into<TagKey>, V: Into<String>>(
+        &mut self,
+        k: K,
+        v: V,
+    ) -> Result<String, TagError>;
 }
 
 impl TagsWrite for Tags {
     fn insert<K: Into<TagKey>, V: Into<String>>(&mut self, k: K, v: V) -> Option<String> {
         self.0.insert(k.into().as_str().to_owned(), v.into())
     }
+    fn checked_insert<K: Into<TagKey>, V: Into<String>>(
+        &mut self,
+        k: K,
+        v: V,
+    ) -> Result<String, TagError> {
+        self.insert(k, v)
+            .ok_or_else(|| TagError("duplicate key".to_owned()))
```

Comment:
I was going to suggest a docstring, but then I got confused over the error case. If `self.insert(k, v).is_some()`, then it's an error. Is this logic backwards or am I misunderstanding the intention?

Reply:
oh yeah, definitely, which reminds me that I also forgot to make use of this code :/